qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -228,34 +260,52 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             if self._sample_from_iterations:
                 power = np.random.randint(power)
             # Run a grover experiment for a given power of the Grover operator.
-            if self._quantum_instance.is_statevector:
-                qc = self.construct_circuit(amplification_problem, power, measurement=False)
-                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+            if self._sampler:
+                qc = self.construct_circuit(amplification_problem, power, measurement=True)
+                job = self._sampler.run([qc])
+
+                # combine the results
```

Comment:
The code seems to just get results from the job, no obvious combining. Maybe just remove the comment?